lib/compress/zstd_compress_internal.h

```diff
@@ -93,6 +97,22 @@ typedef struct {
     U32  staticPrices;           /* prices follow a pre-defined cost structure, statistics are irrelevant */
 } optState_t;
 
+typedef struct {
+    BYTE const* nextSrc;    /* next block here to continue on current prefix */ /* TODO: Does this belong here? */
+    BYTE const* base;       /* All regular indexes relative to this position */
+    BYTE const* dictBase;   /* extDict indexes relative to this position */
+    U32 dictLimit;          /* below that point, need extDict */
+    U32 lowLimit;           /* below that point, no more data */
+    U32 nextToUpdate;       /* index from which to continue table update */
+    U32 nextToUpdate3;      /* index from which to continue table update */
+    U32 hashLog3;           /* dispatch table : larger == faster, more memory */
+    U32 loadedDictEnd;      /* index of end of dictionary */                    /* TODO: Does this belong here? */
```

Comment:
To review.
I don't remember exactly what `loadedDictEnd` does.
My first thought was "it should be `== dictLimit`", but since it exists, that's probably not the case.
Maybe `loadedDictEnd` is really for dictionary loading, while `dictLimit` is a more generic limit for `_extDict` ?
We may have a naming issue here, with different concepts having too similar names.

Reply:
I believe `loadedDictEnd` is for `ZSTD_p_forceWindow`. It is used [here](https://github.com/facebook/zstd/blob/dev/lib/compress/zstd_compress.c#L1716) to either always allow references into the dictionary, or not (if `ZSTD_p_forceWindow == 1`).

Reply:
Right, so this is something set once, on loading dictionary with `ZSTD_p_forceWindow` (or `0` otherwise), and never updated again afterwards.
It's nonetheless useful during compression, to appropriately decide when to update `lowLimit`. So it's part of the compression context.
But it's not part of the "matchfinder" context, since it's never used in matchfinder / parser (while `lowlimit` is).